packages/react-native/React/Fabric/RCTConversions.h

```diff
@@ -34,6 +36,43 @@ inline std::string RCTStringFromNSString(NSString *string)
   return std::string{string.UTF8String ?: ""};
 }
 
+inline UIColor *_Nullable RCTUIColorFromDynamicColor(const facebook::react::Color &color)
+{
+  auto dynamicColor = color.getDynamicColor();
+  int32_t light = dynamicColor.lightColor;
+  int32_t dark = dynamicColor.darkColor;
+  int32_t highContrastLight = dynamicColor.highContrastLightColor;
+  int32_t highContrastDark = dynamicColor.highContrastDarkColor;
+
+  UIColor *lightColor = [RCTConvert UIColor:@(light)];
+  UIColor *darkColor = [RCTConvert UIColor:@(dark)];
+  UIColor *highContrastLightColor = [RCTConvert UIColor:@(highContrastLight)];
+  UIColor *highContrastDarkColor = [RCTConvert UIColor:@(highContrastDark)];
+
+  if (lightColor != nil && darkColor != nil) {
+    UIColor *color = [UIColor colorWithDynamicProvider:^UIColor *_Nonnull(UITraitCollection *_Nonnull collection) {
+      if (collection.userInterfaceStyle == UIUserInterfaceStyleDark) {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastDarkColor != nil) {
+          return highContrastDarkColor;
+        } else {
+          return darkColor;
+        }
+      } else {
+        if (collection.accessibilityContrast == UIAccessibilityContrastHigh && highContrastLightColor != nil) {
+          return highContrastLightColor;
+        } else {
+          return lightColor;
+        }
+      }
+    }];
+    return color;
+  } else {
+    return nil;
+  }
+
+  return nil;
```

Comment:
I think that if a color do not defines both `light` and `darkColor`, the iOS convention is to create a plain color with the only option we have (typically, the default is `light`). 
Can we implement that behavior instead of returning `nil`?

Reply:
Actually, I keep the same code https://github.com/facebook/react-native/blob/b81a081bace78d5bf837d01be5706fffeec50ab8/packages/react-native/React/Base/RCTConvert.m#L948 :) , should we need change that?

Reply:
I see. Let's keep it as it is for the time being. I am not sure that's the canonical way to do it, but at this point there might be libraries or other pieces of code that rely on the fact that this function returns null if the color is not properly setup. :(